api/permissions.py

```diff
@@ -27,3 +29,15 @@ def has_permission(self, request, view):
         assert False, ('TokenHasScope requires either the'
                        '`oauth2_provider.rest_framework.OAuth2Authentication` authentication '
                        'class to be used.')
+
+
+class IsDeletedPremissions(BasePermission):
+    """
+    Permission check for deleted objects.
+    """
+    def has_object_permission(self, request, view, obj):
+        TargetObject = AbstractCreativeWork.objects.filter(id=obj.id).first()
```

Comment:
There no need to reload the object from the database. You can just check `obj.is_deleted`.

Variables should always be lowercase and snakecased, use `target_object` rather than `TargetObject`.